NoClassDefFoundError: com/google/auth/Credentials

I was not signed in to an account when I ran this.

May be a devappserver issue or perhaps an issue with classpath setup. Filing here for the moment.

The servlet that led to this error on a local run is

```
package com.google.fred;

import java.io.IOException;
import java.util.Random;

import javax.servlet.http.HttpServlet;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;

import com.google.cloud.storage.Bucket;
import com.google.cloud.storage.BucketInfo;
import com.google.cloud.storage.Storage;
import com.google.cloud.storage.StorageOptions;

public class HelloAppEngine extends HttpServlet {

  public void doGet(HttpServletRequest request, HttpServletResponse response) 
      throws IOException {
      
	Random r = new Random(System.currentTimeMillis());
	Storage storage = StorageOptions.getDefaultInstance().getService();  
	String bucketName = "my-test-bucket-" + r.nextInt();
	Bucket bucket = storage.create(BucketInfo.of(bucketName));
    response.setContentType("text/plain");
    response.getWriter().print("Created bucket + " + bucketName + "\r\n");

  }
}
```

The classpath was such that this could be compiled.


```
HTTP ERROR 500

Problem accessing /hello. Reason:

    com/google/auth/Credentials
Caused by:

java.lang.NoClassDefFoundError: com/google/auth/Credentials
	at com.google.fred.HelloAppEngine.doGet(HelloAppEngine.java:21)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:617)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:717)
	at org.mortbay.jetty.servlet.ServletHolder.handle(ServletHolder.java:511)
	at org.mortbay.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1166)
	at com.google.appengine.api.socket.dev.DevSocketFilter.doFilter(DevSocketFilter.java:74)
	at org.mortbay.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1157)
	at com.google.appengine.tools.development.devappserver2.RequestIdFilter.doFilter(RequestIdFilter.java:36)
	at org.mortbay.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1157)
	at com.google.appengine.tools.development.ResponseRewriterFilter.doFilter(ResponseRewriterFilter.java:128)
	at org.mortbay.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1157)
	at com.google.appengine.tools.development.HeaderVerificationFilter.doFilter(HeaderVerificationFilter.java:34)
	at org.mortbay.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1157)
	at com.google.apphosting.utils.servlet.TransactionCleanupFilter.doFilter(TransactionCleanupFilter.java:48)
	at org.mortbay.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1157)
	at com.google.appengine.tools.development.StaticFileFilter.doFilter(StaticFileFilter.java:122)
	at org.mortbay.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1157)
	at org.mortbay.jetty.servlet.ServletHandler.handle(ServletHandler.java:388)
	at org.mortbay.jetty.security.SecurityHandler.handle(SecurityHandler.java:216)
	at org.mortbay.jetty.servlet.SessionHandler.handle(SessionHandler.java:182)
	at org.mortbay.jetty.handler.ContextHandler.handle(ContextHandler.java:765)
	at org.mortbay.jetty.webapp.WebAppContext.handle(WebAppContext.java:418)
	at com.google.appengine.tools.development.DevAppEngineWebAppContext.handle(DevAppEngineWebAppContext.java:95)
	at org.mortbay.jetty.handler.HandlerWrapper.handle(HandlerWrapper.java:152)
	at com.google.appengine.tools.development.JettyContainerService$ApiProxyHandler.handle(JettyContainerService.java:508)
	at org.mortbay.jetty.handler.HandlerWrapper.handle(HandlerWrapper.java:152)
	at org.mortbay.jetty.Server.handle(Server.java:326)
	at org.mortbay.jetty.HttpConnection.handleRequest(HttpConnection.java:542)
	at org.mortbay.jetty.HttpConnection$RequestHandler.headerComplete(HttpConnection.java:923)
	at org.mortbay.jetty.HttpParser.parseNext(HttpParser.java:547)
	at org.mortbay.jetty.HttpParser.parseAvailable(HttpParser.java:212)
	at org.mortbay.jetty.HttpConnection.handle(HttpConnection.java:404)
	at org.mortbay.io.nio.SelectChannelEndPoint.run(SelectChannelEndPoint.java:409)
	at org.mortbay.thread.QueuedThreadPool$PoolThread.run(QueuedThreadPool.java:582)
Caused by: java.lang.ClassNotFoundException: com.google.auth.Credentials
	at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at com.google.appengine.tools.development.IsolatedAppClassLoader.loadClass(IsolatedAppClassLoader.java:195)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	... 34 more
Caused by:

java.lang.ClassNotFoundException: com.google.auth.Credentials
	at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at com.google.appengine.tools.development.IsolatedAppClassLoader.loadClass(IsolatedAppClassLoader.java:195)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	at com.google.fred.HelloAppEngine.doGet(HelloAppEngine.java:21)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:617)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:717)
	at org.mortbay.jetty.servlet.ServletHolder.handle(ServletHolder.java:511)
	at org.mortbay.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1166)
	at com.google.appengine.api.socket.dev.DevSocketFilter.doFilter(DevSocketFilter.java:74)
	at org.mortbay.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1157)
	at com.google.appengine.tools.development.devappserver2.RequestIdFilter.doFilter(RequestIdFilter.java:36)
	at org.mortbay.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1157)
	at com.google.appengine.tools.development.ResponseRewriterFilter.doFilter(ResponseRewriterFilter.java:128)
	at org.mortbay.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1157)
	at com.google.appengine.tools.development.HeaderVerificationFilter.doFilter(HeaderVerificationFilter.java:34)
	at org.mortbay.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1157)
	at com.google.apphosting.utils.servlet.TransactionCleanupFilter.doFilter(TransactionCleanupFilter.java:48)
	at org.mortbay.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1157)
	at com.google.appengine.tools.development.StaticFileFilter.doFilter(StaticFileFilter.java:122)
	at org.mortbay.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1157)
	at org.mortbay.jetty.servlet.ServletHandler.handle(ServletHandler.java:388)
	at org.mortbay.jetty.security.SecurityHandler.handle(SecurityHandler.java:216)
	at org.mortbay.jetty.servlet.SessionHandler.handle(SessionHandler.java:182)
	at org.mortbay.jetty.handler.ContextHandler.handle(ContextHandler.java:765)
	at org.mortbay.jetty.webapp.WebAppContext.handle(WebAppContext.java:418)
	at com.google.appengine.tools.development.DevAppEngineWebAppContext.handle(DevAppEngineWebAppContext.java:95)
	at org.mortbay.jetty.handler.HandlerWrapper.handle(HandlerWrapper.java:152)
	at com.google.appengine.tools.development.JettyContainerService$ApiProxyHandler.handle(JettyContainerService.java:508)
	at org.mortbay.jetty.handler.HandlerWrapper.handle(HandlerWrapper.java:152)
	at org.mortbay.jetty.Server.handle(Server.java:326)
	at org.mortbay.jetty.HttpConnection.handleRequest(HttpConnection.java:542)
	at org.mortbay.jetty.HttpConnection$RequestHandler.headerComplete(HttpConnection.java:923)
	at org.mortbay.jetty.HttpParser.parseNext(HttpParser.java:547)
	at org.mortbay.jetty.HttpParser.parseAvailable(HttpParser.java:212)
	at org.mortbay.jetty.HttpConnection.handle(HttpConnection.java:404)
	at org.mortbay.io.nio.SelectChannelEndPoint.run(SelectChannelEndPoint.java:409)
	at org.mortbay.thread.QueuedThreadPool$PoolThread.run(QueuedThreadPool.java:582)
Powered by Jetty://


```

--- a/plugins/com.google.cloud.tools.eclipse.appengine.libraries/plugin.properties
+++ b/plugins/com.google.cloud.tools.eclipse.appengine.libraries/plugin.properties
@@ -2,7 +2,8 @@ pluginName=Cloud Tools For Eclipse App Engine Libraries Management
 providerName=Google Inc.
 objectify.tooltip=com.googlecode.objectify, a Java data access API designed for Google Cloud Datastore	
 appengine.api.tooltip=The com.google.appengine.api, com.google.apphosting, and javax.mail libraries
-googlecloudcore.tooltip=Common functionality required by Google Cloud client libraries including JodaTime
+googlecloudcore.tooltip=Common functionality required by Google Cloud client libraries including \
+ JodaTime, Guava, org.json, and com.google.protobuf
 googlecloudstorage.tooltip=The com.google.cloud.storage packages for client side access to GCS
 googleapiclient.tooltip=Functionality common to Google client APIs, including HTTP transport, \
  error handling, authentication, JSON and protobuf parsing, media download/upload, and batching. 

--- a/plugins/com.google.cloud.tools.eclipse.appengine.libraries/plugin.xml
+++ b/plugins/com.google.cloud.tools.eclipse.appengine.libraries/plugin.xml
@@ -106,39 +106,38 @@
           tooltip="%googleapiclient.tooltip"
           siteUri="https://developers.google.com/api-client-library/java/" >
       <libraryFile
-            javadocUri="https://google.github.io/google-api-java-client/releases/1.21.0/javadoc/index.html">
+          javadocUri="https://google.github.io/google-api-java-client/releases/1.21.0/javadoc/index.html">
         <mavenCoordinates
-              artifactId="google-api-client"
-              groupId="com.google.api-client"
-              version="1.22.0" />
+            artifactId="google-api-client"
+            groupId="com.google.api-client"
+            version="1.22.0" />
       </libraryFile>
       <libraryFile
-            javadocUri="https://google.github.io/google-api-java-client/releases/1.21.0/javadoc/index.html">
+          javadocUri="https://google.github.io/google-api-java-client/releases/1.21.0/javadoc/index.html">
         <mavenCoordinates
-              artifactId="google-api-client-appengine"
-              groupId="com.google.api-client"
-              version="1.22.0" />
+            artifactId="google-api-client-appengine"
+            groupId="com.google.api-client"
+            version="1.22.0" />
+      </libraryFile>
+      <libraryFile>
+        <mavenCoordinates
+            artifactId="google-http-client-appengine"
+            groupId="com.google.http-client"
+            version="1.22.0" />
       </libraryFile>
       <libraryFile
-            javadocUri="https://google.github.io/google-api-java-client/releases/1.21.0/javadoc/index.html">
+          javadocUri="https://google.github.io/google-http-java-client/releases/1.21.0/javadoc/index.html">
         <mavenCoordinates
-              artifactId="google-http-client-appengine"
-              groupId="com.google.http-client"
-              version="1.22.0" />
-      </libraryFile>
-      <libraryFile
-            javadocUri="https://google.github.io/google-http-java-client/releases/1.21.0/javadoc/index.html">
-        <mavenCoordinates
-              artifactId="google-http-client"
-              groupId="com.google.http-client"
-              version="1.22.0" />
+            artifactId="google-http-client"
+            groupId="com.google.http-client"
+            version="1.22.0" />
       </libraryFile>
       <libraryFile javadocUri=
           "https://developers.google.com/api-client-library/java/google-oauth-java-client/reference/1.20.0/">
         <mavenCoordinates
-              artifactId="google-oauth-client"
-              groupId="com.google.oauth-client"
-              version="1.22.0" />
+            artifactId="google-oauth-client"
+            groupId="com.google.oauth-client"
+            version="1.22.0" />
       </libraryFile>
       <libraryFile>  <!-- todo javadoc if it's published -->
         <mavenCoordinates

--- a/plugins/com.google.cloud.tools.eclipse.appengine.libraries/plugin.xml
+++ b/plugins/com.google.cloud.tools.eclipse.appengine.libraries/plugin.xml
@@ -4,6 +4,8 @@
   <extension-point id="com.google.cloud.tools.eclipse.appengine.libraries" 
       name="App Engine Libraries" />
   
+  <!-- todo the dependency tracking is getting ridiculous. Pull this from Maven Central,
+       whether statically or dynamically --> 
   <extension point="com.google.cloud.tools.eclipse.appengine.libraries"
              id="com.google.cloud.tools.eclipse.appengine.libraries.defaultapis">
     <library 
@@ -138,37 +140,33 @@
               groupId="com.google.oauth-client"
               version="1.22.0" />
       </libraryFile>
-      <libraryFile javadocUri=
-          "https://developers.google.com/api-client-library/java/google-oauth-java-client/reference/1.20.0/">
+      <libraryFile>  <!-- todo javadoc if it's published -->
         <mavenCoordinates
               artifactId="google-oauth-client-appengine"
               groupId="com.google.oauth-client"
               version="1.22.0" />
       </libraryFile>      
-      <libraryFile javadocUri=
-          "https://developers.google.com/api-client-library/java/google-oauth-java-client/reference/1.20.0/">
+      <libraryFile>  <!-- todo javadoc if it's published -->
         <mavenCoordinates
               artifactId="google-http-client-jackson2"
               groupId="com.google.http-client"
               version="1.22.0" />
       </libraryFile>
-      <libraryFile javadocUri=
-          "https://developers.google.com/api-client-library/java/google-oauth-java-client/reference/1.20.0/">
+      <libraryFile>  <!-- todo javadoc if it's published -->
         <mavenCoordinates
               artifactId="google-http-client-jackson"
               groupId="com.google.http-client"
               version="1.22.0" />
       </libraryFile>
       <libraryFile javadocUri=
-          "https://developers.google.com/api-client-library/java/google-oauth-java-client/reference/1.20.0/">
+          "http://www.atetric.com/atetric/javadoc/org.codehaus.jackson/jackson-mapper-asl/1.9.13/index.html?overview-summary.html">
         <mavenCoordinates
               artifactId="jackson-core-asl"
               groupId="org.codehaus.jackson"
               version="1.9.13" />
       </libraryFile>
       
-      <libraryFile javadocUri=
-          "https://developers.google.com/api-client-library/java/google-oauth-java-client/reference/1.20.0/">
+      <libraryFile javadocUri="https://fasterxml.github.io/jackson-core/javadoc/2.2.0/">
         <mavenCoordinates
               artifactId="jackson-core"
               groupId="com.fasterxml.jackson.core"
@@ -190,15 +188,13 @@
               groupId="com.google.cloud"
               version="0.10.0-alpha" />
       </libraryFile>
-      <libraryFile
-            javadocUri="https://googlecloudplatform.github.io/google-cloud-java/0.10.0/apidocs/">
+      <libraryFile>  <!-- todo javadoc if it's published -->
         <mavenCoordinates
               artifactId="google-auth-library-oauth2-http"
               groupId="com.google.auth"
               version="0.6.0" />
       </libraryFile>
-      <libraryFile
-            javadocUri="https://googlecloudplatform.github.io/google-cloud-java/0.10.0/apidocs/">
+      <libraryFile> <!-- todo javadoc if it's published -->
         <mavenCoordinates
               artifactId="google-auth-library-credentials"
               groupId="com.google.auth"
@@ -210,36 +206,35 @@
               groupId="joda-time"
               version="2.9.2" />
       </libraryFile>
-       <libraryFile
-            javadocUri="https://googlecloudplatform.github.io/google-cloud-java/0.10.0/apidocs/">
+       <libraryFile javadocUri="https://stleary.github.io/JSON-java/">
         <mavenCoordinates
               artifactId="json"
               groupId="org.json"
               version="20151123" />
       </libraryFile>
       <libraryFile
-            javadocUri="https://googlecloudplatform.github.io/google-cloud-java/0.10.0/apidocs/">
+            javadocUri="https://developers.google.com/protocol-buffers/docs/reference/java/">
         <mavenCoordinates
               artifactId="protobuf-java"
               groupId="com.google.protobuf"
               version="3.0.0" />
       </libraryFile>
       <libraryFile
-            javadocUri="https://googlecloudplatform.github.io/google-cloud-java/0.10.0/apidocs/">
+            javadocUri="https://developers.google.com/protocol-buffers/docs/reference/java/">
         <mavenCoordinates
               artifactId="protobuf-java-util"
               groupId="com.google.protobuf"
               version="3.0.0" />
       </libraryFile>
       <libraryFile
-            javadocUri="https://googlecloudplatform.github.io/google-cloud-java/0.10.0/apidocs/">
+            javadocUri="https://google.github.io/guava/releases/19.0/api/docs/">
         <mavenCoordinates
               artifactId="guava"
               groupId="com.google.guava"
               version="19.0" />
       </libraryFile>
-     <libraryFile
-            javadocUri="https://googlecloudplatform.github.io/google-cloud-java/0.10.0/apidocs/">
+     <!-- todo javadoc -->
+     <libraryFile>
         <mavenCoordinates
               artifactId="gax"
               groupId="com.google.api"
@@ -262,7 +257,7 @@
               version="0.10.0-beta" />
       </libraryFile>
       <libraryFile
-            javadocUri="https://googlecloudplatform.github.io/google-cloud-java/0.10.0/apidocs/">
+            javadocUri="https://developers.google.com/resources/api-libraries/documentation/storage/v1/java/latest/index.html">
         <mavenCoordinates
               artifactId="google-api-services-storage"
               groupId="com.google.apis"

--- a/plugins/com.google.cloud.tools.eclipse.appengine.libraries/plugin.xml
+++ b/plugins/com.google.cloud.tools.eclipse.appengine.libraries/plugin.xml
@@ -111,6 +111,20 @@
               version="1.22.0" />
       </libraryFile>
       <libraryFile
+            javadocUri="https://google.github.io/google-api-java-client/releases/1.21.0/javadoc/index.html">
+        <mavenCoordinates
+              artifactId="google-api-client-appengine"
+              groupId="com.google.api-client"
+              version="1.22.0" />
+      </libraryFile>
+      <libraryFile
+            javadocUri="https://google.github.io/google-api-java-client/releases/1.21.0/javadoc/index.html">
+        <mavenCoordinates
+              artifactId="google-http-client-appengine"
+              groupId="com.google.http-client"
+              version="1.22.0" />
+      </libraryFile>
+      <libraryFile
             javadocUri="https://google.github.io/google-http-java-client/releases/1.21.0/javadoc/index.html">
         <mavenCoordinates
               artifactId="google-http-client"
@@ -123,6 +137,42 @@
               artifactId="google-oauth-client"
               groupId="com.google.oauth-client"
               version="1.22.0" />
+      </libraryFile>
+      <libraryFile javadocUri=
+          "https://developers.google.com/api-client-library/java/google-oauth-java-client/reference/1.20.0/">
+        <mavenCoordinates
+              artifactId="google-oauth-client-appengine"
+              groupId="com.google.oauth-client"
+              version="1.22.0" />
+      </libraryFile>      
+      <libraryFile javadocUri=
+          "https://developers.google.com/api-client-library/java/google-oauth-java-client/reference/1.20.0/">
+        <mavenCoordinates
+              artifactId="google-http-client-jackson2"
+              groupId="com.google.http-client"
+              version="1.22.0" />
+      </libraryFile>
+      <libraryFile javadocUri=
+          "https://developers.google.com/api-client-library/java/google-oauth-java-client/reference/1.20.0/">
+        <mavenCoordinates
+              artifactId="google-http-client-jackson"
+              groupId="com.google.http-client"
+              version="1.22.0" />
+      </libraryFile>
+      <libraryFile javadocUri=
+          "https://developers.google.com/api-client-library/java/google-oauth-java-client/reference/1.20.0/">
+        <mavenCoordinates
+              artifactId="jackson-core-asl"
+              groupId="org.codehaus.jackson"
+              version="1.9.13" />
+      </libraryFile>
+      
+      <libraryFile javadocUri=
+          "https://developers.google.com/api-client-library/java/google-oauth-java-client/reference/1.20.0/">
+        <mavenCoordinates
+              artifactId="jackson-core"
+              groupId="com.fasterxml.jackson.core"
+              version="2.9.0.pr2" />
       </libraryFile>
     </library>
     
@@ -140,11 +190,60 @@
               groupId="com.google.cloud"
               version="0.10.0-alpha" />
       </libraryFile>
+      <libraryFile
+            javadocUri="https://googlecloudplatform.github.io/google-cloud-java/0.10.0/apidocs/">
+        <mavenCoordinates
+              artifactId="google-auth-library-oauth2-http"
+              groupId="com.google.auth"
+              version="0.6.0" />
+      </libraryFile>
+      <libraryFile
+            javadocUri="https://googlecloudplatform.github.io/google-cloud-java/0.10.0/apidocs/">
+        <mavenCoordinates
+              artifactId="google-auth-library-credentials"
+              groupId="com.google.auth"
+              version="0.6.0" />
+      </libraryFile>
       <libraryFile javadocUri="http://www.joda.org/joda-time/apidocs/">
         <mavenCoordinates
               artifactId="joda-time"
               groupId="joda-time"
               version="2.9.2" />
+      </libraryFile>
+       <libraryFile
+            javadocUri="https://googlecloudplatform.github.io/google-cloud-java/0.10.0/apidocs/">
+        <mavenCoordinates
+              artifactId="json"
+              groupId="org.json"
+              version="20151123" />
+      </libraryFile>
+      <libraryFile
+            javadocUri="https://googlecloudplatform.github.io/google-cloud-java/0.10.0/apidocs/">
+        <mavenCoordinates
+              artifactId="protobuf-java"
+              groupId="com.google.protobuf"
+              version="3.0.0" />
+      </libraryFile>
+      <libraryFile
+            javadocUri="https://googlecloudplatform.github.io/google-cloud-java/0.10.0/apidocs/">
+        <mavenCoordinates
+              artifactId="protobuf-java-util"
+              groupId="com.google.protobuf"
+              version="3.0.0" />
+      </libraryFile>
+      <libraryFile
+            javadocUri="https://googlecloudplatform.github.io/google-cloud-java/0.10.0/apidocs/">
+        <mavenCoordinates
+              artifactId="guava"
+              groupId="com.google.guava"
+              version="19.0" />
+      </libraryFile>
+     <libraryFile
+            javadocUri="https://googlecloudplatform.github.io/google-cloud-java/0.10.0/apidocs/">
+        <mavenCoordinates
+              artifactId="gax"
+              groupId="com.google.api"
+              version="0.4.0" />
       </libraryFile>
     </library>
     
@@ -161,6 +260,13 @@
               artifactId="google-cloud-storage"
               groupId="com.google.cloud"
               version="0.10.0-beta" />
+      </libraryFile>
+      <libraryFile
+            javadocUri="https://googlecloudplatform.github.io/google-cloud-java/0.10.0/apidocs/">
+        <mavenCoordinates
+              artifactId="google-api-services-storage"
+              groupId="com.google.apis"
+              version="v1-rev102-1.22.0" />
       </libraryFile>
     </library>
   </extension>

